apps/api/src/app/plugins/bffCache.ts

```diff
@@ -49,43 +54,83 @@ export const bffCache: FastifyPluginCallback<BffCacheOptions> = (
         reply.header(HEADER_NAME, 'HIT');
         reply.type('application/json');
         reply.send(item);
-        return;
+        fastify.log.debug('onRequest cacheItem: %s (%d)', item, ttlInSeconds);
+
+        return reply;
       }
+    } else {
+      fastify.log.trace(`Request not cached for "${key}`);
+      // For now we don't add the cache 'MISS' header, because we only do this for cacheable endpoints (the ones that include cache-control header in the response)
+      // We need to handle this in the "onSend" once we know the actual response
     }
-
-    return;
   });
 
   fastify.addHook('onSend', async function (req, reply, payload) {
-    const isCacheHit = reply.getHeader(HEADER_NAME) === 'HIT';
-    const cacheTtl: number | undefined = getTtlFromResponse(reply, ttl);
-    const isStatus200 = reply.statusCode >= 200 && reply.statusCode < 300;
+    try {
+      const isCacheHit = reply.getHeader(HEADER_NAME) === 'HIT';
+      const cacheTtl: number | undefined = getTtlFromResponse(reply, ttl);
+      const isStatus200 = reply.statusCode >= 200 && reply.statusCode < 300;
+
+      // If the cache is a hit, or is non-cacheable, we just proceed with the request
+      if (isCacheHit || cacheTtl === undefined || !isStatus200) {
+        return undefined;
+      }
 
-    // If the cache is a hit, or is non-cacheable, we just proceed with the request
-    if (isCacheHit || cacheTtl === undefined || !isStatus200) {
-      return;
+      // If there is no cached data, then its a cache-miss
+      reply.header(HEADER_NAME, 'MISS');
+
+      // Get content from payload
+      const content =
+        typeof payload === 'string'
+          ? payload
```

Comment:
don't ask me why, but this part is important that if the payload is a string, we do this code synchronously. 

If we await for the payload (for example, by moving this typeof check to `getContentFromPayload`, then we get an error).

This took me a lot of time to debug! 🤯